cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3530,11 +3708,26 @@ private boolean isCustomCertUpToDate(StatefulSet sts, Pod pod, String annotation
                 return new ArrayList<>();
             }
 
-            boolean isPodUpToDate = isPodUpToDate(sts, pod);
-            boolean areCustomListenerCertsUpToDate = isCustomCertUpToDate(sts, pod, KafkaCluster.ANNO_STRIMZI_CUSTOM_LISTENER_CERT_THUMBPRINTS);
-
             List<String> reasons = new ArrayList<>(3);
 
+            if (ctrlResource instanceof StatefulSet) {
+                StatefulSet sts = (StatefulSet) ctrlResource;
+
+                if (!isPodUpToDate(sts, pod)) {
+                    reasons.add("Pod has old generation");
+                }
+
+                if (!isCustomCertUpToDate(sts, pod, KafkaCluster.ANNO_STRIMZI_CUSTOM_LISTENER_CERT_THUMBPRINTS)) {
+                    reasons.add("custom certificate one or more listeners changed");
```

Comment:
At this point, we support only ZooKeeper. And ZooKeeper has no custom listener certificates. I expect that this will need to be handled when we add support for Kafka. But adding it now makes little sense since it cannot be tested etc.